rust/cloud-storage/properties/src/outbound/permission_service.rs

```diff
@@ -82,4 +83,41 @@ impl PermissionService for PermissionServiceImpl {
             Some(_) | None => anyhow::bail!("Access denied"),
         }
     }
+
+    #[tracing::instrument(skip(self), fields(entity_id = %entity_id, entity_type = ?entity_type, user_count = user_ids.len()), err)]
+    async fn grant_entity_permissions(
+        &self,
+        user_ids: &[String],
+        entity_id: &str,
+        entity_type: EntityType,
+    ) -> Result<(), Self::Err> {
+        if user_ids.is_empty() {
+            return Ok(());
```

Comment:
Shouldn't this be called `grant_permissions_to_task` it only handles tasks right ?